src/data/terraform.json

```diff
@@ -21,40 +21,61 @@
 		"plugin/sdkv2",
 		"registry"
 	],
+	"docsNavItems": [
```

Comment:
A new property on products. _Very_ open to moving this around, renaming it, etc.

Intent here is that we can meet the needs of products like Terraform, where not all `rootDocsPaths` are meant to be shown in navs, and additional custom items are needed.

The initial approach I considered was to add some property to `rootDocsPaths`, like `showInNav=false` or something. However, this does not account for custom items such as Terraform's links to `/terraform/language/providers` and `/terraform/docs/partnerships`.